peps/pep-9999.rst

```diff
@@ -64,34 +73,110 @@ Open Issues
+
+Polling shows ``intmath`` as another popular name.  The argument made was that
+the normal mathematical spelling of the imaginary unit is i, which makes imath
+ambiguous.  It also has no conflict with any PyPI module.  On the other hand,
+perhaps ``intmath`` may be confused with interval math or numerical
+integration.
+
+Other proposed names include ``ntheory`` (like SymPy's submodule),
 ``integermath`` and ``imaths``.
 
-Possible renaming of isqrt
---------------------------
+Module scope and possible library members
+=========================================
+
+The following is a list of functions that could one day find a home in the
+``imath`` library.  New fucntions are not part of the proposal.  They
+illustrate the proposed scope of ``imath`` and its potential utility.
+Functions proposed in the initial discussion thread include:
+
+* Divisibility & Modular Arithmetic (see also [5]_)
+
+  * ``gcd(a, b)``: Already in ``math``.
+  * ``lcm(a, b)``: Already in ``math``.
+  * ``powmod(a, b, m)``: Modular exponentiation (already in pow(a, b, m), but
+    renamed for clarity).
+  * ``is_coprime(a, b)``
+  * ``modinv(a, m)``: Modular inverse (or perhaps better yet, ``gcdext``
+    below).
+  * ``gcdext()`` --- to solve linear Diophantine equation in two variables (the
+    :external+py3.14:class:`int` implementation actually includes extended
+    Euclidean algorithm).
+
+* Integer Arithmetic
+
+  * ``ceil_div(a, b)``: Ceiling divide, see [2]_, [3]_.
+  * ``ilog(n, base)``: Integer logarihm; currently
+    :external+py3.14:func:`math.log` has a special handling for integer
+    arguments.  It's unique (wrt other module functions) and not documented so
+    far. See [4]_.
+
+* Combinatorics and Integer Sequences
+
+  * ``factorial(n)``: Already in ``math``.
+  * ``comb(n, k)``: Already in ``math``.
+  * ``fibonacci(n)``
+
+* Integer Roots
+
+  * ``isqrt(n)``: Already in ``math``.
+  * ``isqrt_rem(n)``: The integer square root and its remainder.
+  * ``sqrtmod(n, modulo)``: Square root within a modular group.
+
+* Primality & Factorization
+
+  * ``is_prime(n)``: Optimized for large numbers.
+  * ``prime_factors(n)``: Trial division or Pollard’s rho for small inputs.
+  * ``primes_between(lower, upper)``: Uses the sieve of Atkin for large ranges
+    that start near zero, or an iterative algorithm for small ranges that are
+    far from zero.
+  * ``totient(n)``: Euler’s totient function.
+  * ``next_prime(n)``
+  * ``prev_prime(n)``
+
+* Encoding & Number Representations
+
+  * ``gray_encode(n), gray_decode(n)``
+  * ``hamming_weight(n)``
+  * ``hamming_distance(a, b)``
+
+* Binary Arithmetic
+
+  * ``is_power_of_two(x)``
+  * ``next_power_of_two(x)``
+
+
+Rejected ideas
+==============
+
+There was a brief discussion about exposing ``math.isqrt`` as ``imath.sqrt``.
+However, ``isqrt`` is ultimately a different function: it is the floor of the
+square root.  It would be confusing to give it the same name (under a different
+module).
```

Comment:
```suggestion
There was a brief discussion about exposing :external+py3.14:func:`math.isqrt`
as ``imath.sqrt``.  However, ``isqrt()`` is ultimately a different function:
it is the floor of the square root.  It would be confusing to give it the same name
(under a different module) as :external+py3.14:func:`math.sqrt` or
:external+py3.14:func:`cmath.sqrt`.
```

Reply:
Done.